frontend/tickets-online/src/hall/containers/HallContainer.js

```diff
@@ -76,20 +87,80 @@ class HallModelContainer extends React.Component {
     }
   }
 
+  onServicesClick(operation, serviceId) {
+    let newServices = [];
+    let t = this;
+    switch (operation) {
+      case 'plus':
+        newServices = this.state.services.map(function(elem) {
+          if (elem.id === serviceId) {
+            elem.amount += 1;
+            t.setState({
+              servicesResultPrice: t.state.servicesResultPrice + elem.price
+            });
+            return elem;
+          }
+          return elem;
+        });
+        break;
+      case 'minus':
+        newServices = this.state.services.map(function(elem) {
+          if (elem.id === serviceId) {
+            if (elem.amount >= 1) {
+              elem.amount -= 1;
+              t.setState({
+                servicesResultPrice: t.state.servicesResultPrice - elem.price
+              });
+            } else {
+              elem.amount = 0;
+            }
+            return elem;
+          }
+          return elem;
+        });
+        break;
+      default:
+        break;
+    }
+    this.setState({
+      services: newServices
+    });
+  }
+
   onBookBtnClick() {
-    // placePriceId -> setTicket and go to the ticketComponent
-    console.log(this.state.placeInfo.placePriceId);
+    let servicesArr = [];
+    this.state.services.forEach(function(elem) {
+      if (elem.amount !== 0) {
+        servicesArr.push({
+          id: elem.id,
+          amount: elem.amount
+        });
+      }
+    });
+    let newTicket = {
+      priceId: this.state.placeInfo.placePriceId,
+      services: servicesArr.length !== 0 ? servicesArr : null
+    };
+    console.log(newTicket);
```

Comment:
Please do not forget to delete such messages. Also it is worth to use console.debug that won't be called if dev tools are closed.